orion/sites/admin/src/home/home.js

```diff
@@ -2,30 +2,195 @@
 require("./home.styl");
 import React from "react";
 import ReactDOM from "react-dom";
+import API from "../api.js";
 
 export class HomePage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            programs: [],
+            classes: [],
+            locations: [],
+            achieves: [],
+            semesters: [],
+            sessions: [],
+            rows: [],
+            selection: "",
+            noUnpub: "",
+            switch: false,
+        };
+        this.onClickPrograms = this.onClickPrograms.bind(this);
+        this.onClickClasses = this.onClickClasses.bind(this);
+        this.onClickLocations = this.onClickLocations.bind(this);
+        this.onClickAchievements = this.onClickAchievements.bind(this);
+        this.onClickSemesters = this.onClickSemesters.bind(this);
+        this.onClickSessions = this.onClickSessions.bind(this);
+        this.onClickPage = this.onClickPage.bind(this);
+    }
+
+    componentDidMount() {
+        API.get("api/v1/unpublished").then((res) => {
+            const unpublishedList = res.data;
+            this.setState({
+                programs: unpublishedList.programs,
+                classes: unpublishedList.classes,
+                locations: unpublishedList.locations,
+                achieves: unpublishedList.achieves,
+                semesters: unpublishedList.semesters,
+                sessions: unpublishedList.sessions,
+            });
+        });
+    }
+
+    onClickPrograms() {
+        let programs = this.state.programs.map((row, index) => {
+            return <DashboardRow key={index} title={row.name} />;
+        });
+        this.setState({
+            noUnpub:
+                programs.length == 0
+                    ? "All items are published! Your website is up to date!"
+                    : "",
+            rows: programs,
+            selection: "programs",
+            switch: programs.length == 0 ? false : true,
```

Comment:
I guess technically this can just be `switch: programs.length != 0` but at least you got try out the `? :` operator!